addons/binding/org.openhab.binding.fritzaha/src/main/java/org/openhab/binding/fritzaha/handler/FritzAhaBridgeHandler.java

```diff
@@ -0,0 +1,282 @@
+		}
+	}
+
+	/**
+	 * Disposes the bridge.
+	 */
+	@Override
+	public void dispose() {
+		LOG.debug("Handler disposed.");
+		if (pollingJob != null && !pollingJob.isCancelled()) {
+			pollingJob.cancel(true);
+			pollingJob = null;
+		}
+	}
+
+	/**
+	 * Called from {@link FritzAhaUpdateXmlCallback} to provide new values for
+	 * things.
+	 * 
+	 * @param model
+	 *            Device model with updated data.
+	 */
+	public void addDeviceList(DeviceModel model) {
+		try {
+			LOG.debug("set device model: " + model.toString());
+			this.deviceList.put(model.getIdentifier(), model);
+			ThingUID thingUID = this.getThingUID(model);
+			LOG.debug("update thing " + thingUID + " with device model: "
+					+ model.toString());
+			this.updateThingFromDevice(this.getThingByUID(thingUID), model);
+		} catch (Exception e) {
+			LOG.error(e.getLocalizedMessage(), e);
+		}
+	}
+
+	/**
+	 * Provides the web interface object.
+	 * 
+	 * @return
+	 */
+	public FritzahaWebInterface getWebInterface() {
+		return this.connection;
+	}
+
+	/**
+	 * Updates things from device model.
+	 * 
+	 * @param thing
+	 *            Thing to be updated.
+	 * @param device
+	 *            Device model with new data.
+	 */
+	private void updateThingFromDevice(Thing thing, DeviceModel device) {
+		if (thing == null || device == null) {
+			LOG.error("thing or device null, cannot perform update");
+			return;
+		}
+		LOG.debug("about to update " + thing.getUID() + " from "
+				+ device.toString());
+		if (device.isTempSensor()) {
+			Channel channel = thing.getChannel(CHANNEL_TEMP);
+			this.updateState(channel.getUID(), new DecimalType(device
+					.getTemperature().getCelsius()));
+		}
+		if (device.isPowermeter()) {
+			Channel channelEnergy = thing.getChannel(CHANNEL_ENERGY);
+			this.updateState(channelEnergy.getUID(), new DecimalType(device
+					.getPowermeter().getEnergy()));
+			Channel channelPower = thing.getChannel(CHANNEL_POWER);
+			this.updateState(channelPower.getUID(), new DecimalType(device
+					.getPowermeter().getPower()));
+		}
+		if (device.isSwitchableOutlet()) {
+			Channel channel = thing.getChannel(CHANNEL_SWITCH);
+			if (device.getSwitch().getState().equals(SwitchModel.ON)) {
+				this.updateState(channel.getUID(), OnOffType.ON);
+			} else if (device.getSwitch().getState().equals(SwitchModel.OFF)) {
+				this.updateState(channel.getUID(), OnOffType.OFF);
+			} else {
+				LOG.error("unknown state " + device.getSwitch().getState()
+						+ " for channel " + channel.getUID());
+			}
+		}
+	}
+
+	/**
+	 * Builds a {@link ThingUID} from a device model.
+	 * 
+	 * @param device
+	 * @return
+	 */
+	public ThingUID getThingUID(DeviceModel device) {
+		ThingUID bridgeUID = this.getThing().getUID();
+		ThingTypeUID thingTypeUID = new ThingTypeUID(BINDING_ID, device
+				.getProductName().replaceAll("[^a-zA-Z0-9_]", "_"));
+
+		if (FritzAhaHandler.SUPPORTED_THING_TYPES.contains(thingTypeUID)) {
+			String thingName = device.getName()
```

Comment:
Is device.getName() a unique identifier (like the AIN)? Or will this change, if the user renames his device in the Fritz!Box?